hotel/src/hil/rng.rs

```diff
@@ -0,0 +1,25 @@
+/// A Random Number Generator, based on the Rng trait from the rand crate.
+pub trait Rng {
+    /// Returns 4 bytes of random data.  Types that implement the Rng trait
+    /// only need to implement this function.
+    fn next_u32(&mut self) -> u32;
```

Comment:
@jynnantonix I sketched out an suggestion for an interface that, I believe, maps to _nearly_ identical operation in the hotel case, but gracefully falls-back to asynchronous I/O when needed.

The trait is here: https://github.com/alevy/tock/blob/hil/rng/kernel/src/hil/rng.rs
and an initial sketch implementation for the SAM4L is here: https://github.com/alevy/tock/blob/hil/rng/chips/sam4l/src/trng.rs

The basic idea is that on every interrupt, instead of returning just a single u32 up to the client, or requiring the client to pass in a statically allocated buffer (as is done in the UART), the RNG passes up an Iterator, which the client can use to consume as many values as needed and available. When the iterator returns None, it means there are no more values currently available, and the client can return `Continue::More` to tell the RNG that it wants to be called again when more data is available.

Feel free to use/not use as you please. Note that this is _just a sketch_, not even a functional prototype yet.